include/pluginlib/class_loader_imp.h

```diff
@@ -1,806 +1,36 @@
-  }
-}
-
-template<class T>
-void ClassLoader<T>::refreshDeclaredClasses()
-/***************************************************************************/
-{
-  ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Refreshing declared classes.");
-  // determine classes not currently loaded for removal
-  std::list<std::string> remove_classes;
-  for (std::map<std::string, ClassDesc>::const_iterator it = classes_available_.begin();
-    it != classes_available_.end(); it++)
-  {
-    std::string resolved_library_path = it->second.resolved_library_path_;
-    std::vector<std::string> open_libs = lowlevel_class_loader_.getRegisteredLibraries();
-    if (std::find(open_libs.begin(), open_libs.end(), resolved_library_path) != open_libs.end()) {
-      remove_classes.push_back(it->first);
-    }
-  }
-
-  while (!remove_classes.empty()) {
-    classes_available_.erase(remove_classes.front());
-    remove_classes.pop_front();
-  }
-
-  // add new classes
-  plugin_xml_paths_ = getPluginXmlPaths(package_, attrib_name_, true);
-  std::map<std::string, ClassDesc> updated_classes = determineAvailableClasses(plugin_xml_paths_);
-  for (std::map<std::string, ClassDesc>::const_iterator it = updated_classes.begin();
-    it != updated_classes.end(); it++)
-  {
-    if (classes_available_.find(it->first) == classes_available_.end()) {
-      classes_available_.insert(std::pair<std::string, ClassDesc>(it->first, it->second));
-    }
-  }
-}
-
-template<class T>
-std::string ClassLoader<T>::stripAllButFileFromPath(const std::string & path)
-/***************************************************************************/
-{
-  std::string only_file;
-  size_t c = path.find_last_of(getPathSeparator());
-  if (std::string::npos == c) {
-    return path;
-  } else {
-    return path.substr(c, path.size());
-  }
-}
-
-template<class T>
-int ClassLoader<T>::unloadLibraryForClass(const std::string & lookup_name)
-/***************************************************************************/
-{
-  ClassMapIterator it = classes_available_.find(lookup_name);
-  if (it != classes_available_.end() && it->second.resolved_library_path_ != "UNRESOLVED") {
-    std::string library_path = it->second.resolved_library_path_;
-    ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Attempting to unload library %s for class %s",
-      library_path.c_str(), lookup_name.c_str());
-    return unloadClassLibraryInternal(library_path);
-  } else {
-    throw pluginlib::LibraryUnloadException(getErrorStringForUnknownClass(lookup_name));
-  }
-}
-
-template<class T>
-int ClassLoader<T>::unloadClassLibraryInternal(const std::string & library_path)
-/***************************************************************************/
-{
-  return lowlevel_class_loader_.unloadLibrary(library_path);
-}
-
-}  // namespace pluginlib
+#error Header <pluginlib/class_loader_imp.h> is deprecated, \
+but this file should not be included directly, include <pluginlib/class_loader.hpp> instead.
```

Comment:
cpplint being unable to figure out that this is a message. It complains about the use of `not` here. We could either replace it with "never" or silence cpplint on this line